core/src/main/java/org/geysermc/geyser/translator/protocol/java/JavaKeepAliveTranslator.java

```diff
@@ -42,10 +43,11 @@ public void translate(GeyserSession session, ClientboundKeepAlivePacket packet)
         if (!session.getGeyser().getConfig().isForwardPlayerPing()) {
             return;
         }
-        // We use this once the client replies (see BedrockNetworkStackLatencyTranslator)
-        session.getKeepAliveCache().add(packet.getPingId());
 
-        long timestamp = packet.getPingId();
+        // We use this once the client replies
+        final long javaId = packet.getPingId();
+
+        long timestamp = javaId;
```

Comment:
All of the timestamp handling below this can be probably removed if we're no longer relying on the sign of the bedrock response timestamp being consistent